### DISCUSSION TITLE
Warranty Life Spent Calculations

### DISCUSSION THREAD

**Discussion by fbradyirl** [TypQxQ/Sigenergy-Home-Assistant-Integration#7] [General]
I'm not so sure about the current calculations regarding battery warranty. You seem to be adding charge + discharge to obtain a total amount. However, I think with Sigenergy that only the discharge amount is relevant (I could be wrong). 

<img width="513" alt="Screenshot 2024-09-16 at 15 12 17" src="https://github.com/user-attachments/assets/b4c40f62-c162-4354-b2a9-e195fbcd8e39">

When this value reaches the threshold the warranty is spent (or 10 years elapses).

**Comment by Wictorssons** [comment]
Also, I have not seen anything regarding charge/discharge in the warranty information. Only 10 years is mentioned as far as I have seen.
Have you seen otherwise?

**Reply by fbradyirl** [reply]
<img width="1317" alt="image" src="https://github.com/user-attachments/assets/b18e1448-55be-4b04-a212-de504ed13bbf">

https://www.sigenergy.com/uploads/en_download/1721976954430447.pdf?fbclid=IwY2xjawFVH5xleHRuA2FlbQIxMAABHc1pSzXz_flqJGC5C1iiXK8Vp1F8jv9ibOMATv4vYmVN8i18CR7PBB25tg_aem_QrVCZCpv_njF3o6ATUXNPg

**Reply by fbradyirl** [reply]
![image](https://github.com/user-attachments/assets/2feddb58-a2df-4995-8d84-3dd4f41c94e3)

Confirmed with Sigenergy support.

**Comment by fbradyirl** [comment]
Discussion on same: https://www.facebook.com/groups/sigenergy/permalink/533502209018865/

**Comment by TypQxQ** [comment]
This is great news, Thank you!
I changed it right away 😃
